src/extensions/texter-sideboxes/contact-notes/react-component.js

```diff
@@ -0,0 +1,164 @@
+import PropTypes from "prop-types";
+import React from "react";
+import { withRouter } from "react-router";
+import gql from "graphql-tag";
+import _ from "lodash";
+
+import TextField from "@material-ui/core/TextField";
+import CircularProgress from "@material-ui/core/CircularProgress";
+
+import loadData from "../../../containers/hoc/load-data";
+
+export const displayName = () => "Contact notes";
+
+export const showSidebox = () => true;
+
+export class TexterSideboxClass extends React.Component {
+  constructor(props) {
+    /* eslint-disable no-param-reassign */
+    if (typeof props.contact.customFields === "string") {
+      try {
+        props.contact.customFields = JSON.parse(props.contact.customFields);
```

Comment:
Is this block still necessary re: https://github.com/MoveOnOrg/Spoke/pull/2192#discussion_r917423549?

Reply:
Yep this is what makes the other one unnecessary. I realized other places in the client expect the `customFields` to be stringified, so the conversion should happen only here rather than on the server.

That said, [I discovered and fixed a regression here](https://github.com/MoveOnOrg/Spoke/pull/2192/commits/47df5ca30e1bf4267dc6d48e699615d0c190fffa) – I was mutating the `customFields` in the props like I had it which was causing issues in other places that expected it to be a string. A reminder to myself never to directly reassign values to props or state objects in React, always create a copy!